### PR TITLE
Implement JWT Authentication and Role Configuration

### PR DESCRIPTION
**Pull request by chertiav**
- added `AuthenticationService`;
- added `JwtAuthenticationFilter`;
- added `JwtUtil`;
- integrated login functionality in `AuthController`;
- refactored `UserMapper` for role assignment.

### CODE REVIEW

File: src/main/java/com/achdev/onlinebookstoreapp/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/config/OpenApiConfig.java]: create constants for strings

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]: create constant for "Bearer "

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(constant.length());
```

File: src/main/java/com/achdev/onlinebookstoreapp/mapper/UserMapper.java [hunk 5]
```diff
@@ -3,16 +3,34 @@
...
+    protected RoleRepository roleRepository;
+
+    public abstract UserDto toDto(User user);
 
     @Mapping(source = "password", target = "password", qualifiedBy = EncodedMapping.class)
-    User toModel(UserRegistrationRequestDto requestDto);
+    public abstract User toModel(UserRegistrationRequestDto requestDto);
+
+    @AfterMapping
+    public void setDefaultRole(@MappingTarget User user) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/mapper/UserMapper.java]: this approach could cause some problems
this is business logic that should be in service layer
hard to testing
and others

let's replace this logic to service

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/achdev/onlinebookstoreapp/config/OpenApiConfig.java (hunk 1 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/config/OpenApiConfig.java]: Ok

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java (hunk 2 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]: Ok

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java (hunk 3 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]: Ok

File: src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java (hunk 4 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/security/JwtAuthenticationFilter.java]: Ok

File: src/main/java/com/achdev/onlinebookstoreapp/mapper/UserMapper.java (hunk 5 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/mapper/UserMapper.java]: Ok

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**